pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -583,15 +583,31 @@ private void createDerivedColumnV1Indices(String column, FunctionEvaluator funct
       Object[] outputValues = new Object[numDocs];
       PinotDataType outputValueType = null;
       for (int i = 0; i < numDocs; i++) {
+        boolean hasNull = false;
         for (int j = 0; j < numArguments; j++) {
           inputValues[j] = valueReaders.get(j).getValue(i);
+          Object defaultNullValue = argumentsMetadata.get(j).getFieldSpec().getDefaultNullValue();
+          if (inputValues[j] == null || inputValues[j] == defaultNullValue) {
```

Comment:
I don't think this check is necessary. When transform fails and `errorOnFailure` is false, we can put `null` value

Reply:
done